src/service/viewer-impl.js

```diff
@@ -130,6 +130,9 @@ export class Viewer {
     /** @private {!Observable<!JSONType>} */
     this.broadcastObservable_ = new Observable();
 
+    /** @private {!Observable<!JSONType>} */
+    this.scrollDocObservable_ = new Observable();
```

Comment:
There'd be a map of observables: `!Object<string, !Observable<!JSONType>>`. The index is the type of message. The API will be `viewer.onMessage(eventType, handler)`. The records in the map will be created from calls to this API. Thus we simply remove a switch `if (eventType == 'x') {x.fire(payload);}` to simply `if (observables[eventType]) {observables[eventType].fire(payload)}`.